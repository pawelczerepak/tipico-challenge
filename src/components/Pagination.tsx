import React from 'react';
import styled from 'styled-components';
import { uniq } from 'lodash';

interface Props {
    pageCount: number;
    currentPage: number;
    onChangePage: (page: number) => void;
}

const Wrapper = styled.ul`
  list-style-type: none;
  text-align: center;
`;

const Page = styled.li`
  display: inline-block;
  min-width: 40px;
  text-align: center;
  border: 1px solid #ced4da;
  cursor: pointer;
  padding: 5px;
  background-color: ${({isCurrent}: {isCurrent: any}) => isCurrent ? '#c8102e' : 'transparent'};
  color: ${({isCurrent}: {isCurrent: any}) => isCurrent ? '#fff' : 'initial'};
  &:not(:first-child) {
    border-left: none;
  }
`;

export default ({pageCount, currentPage, onChangePage}: Props) => {
    const pages = uniq([1, Math.max(1, currentPage - 1), currentPage, Math.min(pageCount, currentPage + 1), pageCount]);

    const makePage = (page: number) => {
        const onClick = () => onChangePage(page);
        return <Page key={page} onClick={onClick} isCurrent={page === currentPage}>{page}</Page>;
    };
    return <Wrapper>
        {pages.map(makePage)}
    </Wrapper>;
};
