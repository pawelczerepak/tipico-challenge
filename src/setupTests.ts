import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';
import client from './apiClient';

configure({ adapter: new Adapter() });

export const mockStore = configureMockStore([thunk, axiosMiddleware(client)]);
