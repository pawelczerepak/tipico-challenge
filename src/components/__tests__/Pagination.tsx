import React from 'react';
import '../../setupTests';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from '../Pagination';

describe('Link', () => {
    it('capture snapshot', () => {
        expect(toJson(shallow(
            <Pagination pageCount={5} currentPage={1} onChangePage={jest.fn()}/>,
        )))
            .toMatchSnapshot();
    });
});
