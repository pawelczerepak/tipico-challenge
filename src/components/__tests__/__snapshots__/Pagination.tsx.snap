// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link capture snapshot 1`] = `
<styled.ul>
  <styled.li
    isCurrent={true}
    key="1"
    onClick={[Function]}
  >
    1
  </styled.li>
  <styled.li
    isCurrent={false}
    key="2"
    onClick={[Function]}
  >
    2
  </styled.li>
  <styled.li
    isCurrent={false}
    key="5"
    onClick={[Function]}
  >
    5
  </styled.li>
</styled.ul>
`;
